stdlib/public/core/DoubleWidth.swift.gyb

```diff
@@ -531,7 +536,12 @@ extension DoubleWidth : FixedWidthInteger {
   
   @_inlineable // FIXME(sil-serialize-all)
   public static func &>>=(lhs: inout DoubleWidth, rhs: DoubleWidth) {
-    let rhs = rhs & DoubleWidth(DoubleWidth.bitWidth &- 1)
+    // FIXME(integers): test types with bit widths that aren't powers of 2
+    let rhs = DoubleWidth.bitWidth.nonzeroBitCount == 1
```

Comment:
This *masking* piece should also be extracted into a function, I think, which can have an additional benefit of a descriptive name.